webpack-dll.config.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External Dependencies
+ */
+const path = require( 'path' );
+const webpack = require( 'webpack' );
+
+/**
+ * Internal Dependencies
+ */
+const config = require( './server/config' );
+
+const bundleEnv = config( 'env' );
+
+module.exports = {
+	entry: {
+		vendor: [ path.join( __dirname, 'client', 'vendor-dll.js' ) ]
+	},
+	output: {
+		path: path.join( __dirname, 'public' ),
+		publicPath: '/calypso/',
+		filename: '[name].' + bundleEnv + '.js',
+		library: '[name]',
+		devtoolModuleFilenameTemplate: 'app:///[resource-path]'
+	},
+	plugins: [
+		new webpack.DllPlugin( {
+			path: path.join( __dirname, 'build', 'dll', '[name].' + bundleEnv + '-manifest.json' ),
+			name: '[name]',
+			context: path.resolve( __dirname, 'client' )
+		} ),
+		new webpack.DefinePlugin( {
+			'process.env': {
+				NODE_ENV: JSON.stringify( bundleEnv )
+			}
+		} ),
+		new webpack.optimize.OccurenceOrderPlugin()
+	],
+	module: {
+		loaders: [
+			{
+				test: /\.json$/,
+				loader: 'json-loader'
+			},
+			{
+				test: /\.html$/,
+				loader: 'html-loader'
+			}
+		]
+	},
+	node: {
+		console: false,
+		process: true,
+		global: true,
+		Buffer: true,
+		__filename: 'mock',
+		__dirname: 'mock',
+		fs: 'empty'
+	},
+	resolve: {
+		root: path.resolve( __dirname, 'client' )
+	}
+};
```

Comment:
At some point we should consider extracting out common bits across all our Webpack configs and having individual ones extend via e.g. `_.merge`.